src/controller/python/chip/ChipStack.py

```diff
@@ -151,10 +151,14 @@ def __call__(self):
             self._cv.notify_all()
         pythonapi.Py_DecRef(py_object(self))
 
-    def Wait(self):
+    def Wait(self, timeoutMs: int = None):
+        timeout = None
+        if timeoutMs is not None:
+            timeout = float(timeoutMs) / 1000
+
         with self._cv:
             while self._finish is False:
-                self._cv.wait()
+                self._cv.wait(timeout)
```

Comment:
Shouldn't this check the return value?

Reply:
Good point. Yes it should. I'll raise an exception if it returns false.